packages/controllers/src/snaps/SnapController.ts

```diff
@@ -1034,7 +1039,29 @@ export class SnapController extends BaseController<
       }
 
       this._recordSnapRpcRequest(snapName);
-      return handler(origin, request);
+
+      // handle max request time
+      let resolve: any;
+      let reject: any;
+
+      const timeoutPromise = new Promise((res, rej) => {
+        resolve = res;
+        reject = rej;
+      });
+
+      const timeout = window.setTimeout(() => {
+        this._transitionSnapState(snapName, SnapStatusEvent.stop);
+        this._stopSnap(snapName, false);
```

Comment:
Why don't we just not pass `false` as the second parameter to `_stopSnap` and let it handle the state transition?